core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/TrackNetworkInfra.kt

```diff
@@ -67,3 +69,8 @@ typealias EndpointTrackSectionId = EndpointStaticIdx<TrackSection>
 typealias OptEndpointTrackSectionId = OptEndpointStaticIdx<TrackSection>
 val TrackSectionId.start get() = EndpointTrackSectionId(this, Endpoint.START)
 val TrackSectionId.end get() = EndpointTrackSectionId(this, Endpoint.END)
+
+@JvmName("getTrackEndpoint")
+fun getTrackEndpoint(trackSection: DirTrackSectionId): EndpointTrackSectionId {
+    return EndpointTrackSectionId(trackSection.value, trackSection.direction.toEndpoint);
+}
```

Comment:
This change helped me contemplate relevance of `EndpointTrackSectionId`s, at least for this usecase. If endpoints can be converted to directions along a track, what's the point? I also think endpoints are more confusing to work with than directions along a track in this case.

I would consider removing all `EndpointTrackSectionId`s, and switching to `DirTrackSectionId`s.

Reply:
As discussed, it will be clearer if we keep the `EndpointTrackSectionId`s for all the methods related to `ports` and `nodes`. However, `DirTrackSectionId`s will be used for the `getNextTrackSection` and `getNextTrackSections` methods.